lib/chef/resource/windows_firewall_profile.rb

```diff
@@ -0,0 +1,168 @@
+      ```
+
+      **Enable and Configure the Public Profile of the Windows Firewall**:
+
+      ```ruby
+      windows_firewall_profile 'Configure and Enable Windows Firewall Public Profile' do
+        profile 'Public'
+        profile_enabled true
+        default_inbound_block true
+        default_outbound_allow true
+        allow_inbound_rules false
+        display_notification false
+        action :configure
+      end
+      ```
+      **Disable the Domain Profile of the Windows Firewall**:
+
+      ```ruby
+      windows_firewall_profile 'Disable the Domain Profile of the Windows Firewall' do
+        profile 'Domain'
+        profile_enabled false
+        action :configure
+      end
+      ```
+      DOC
+
+      unified_mode true
+
+      property :profile, String, required: true, equal_to: %w{ Domain Public Private }, description: "Set the Windows Profile being configured"
+      property :profile_enabled, [true, false], default: true, description: "Set the status of the firewall profile to Enabled or Disabled"
+      property :default_inbound_block, [true, false, nil], default: true, description: "Set the default policy for inbound network traffic to blocked"
+      property :default_outbound_allow, [true, false, nil], default: true, description: "Set the default policy for outbound network traffic to allowed"
+      property :allow_inbound_rules, [true, false, nil], description: "Allow users to set inbound firewall rules"
+      property :allow_local_firewall_rules, [true, false, nil], description: "Merges inbound firewall rules into the policy"
+      property :allow_local_ipsec_rules, [true, false, nil], description: "Allow users to manage local connection security rules"
+      property :allow_user_apps, [true, false, nil], description: "Allow user applications to manage firewall"
+      property :allow_user_ports, [true, false, nil], description: "Allow users to manage firewall port rules"
+      property :allow_unicast_response, [true, false, nil], description: "Allow unicast responses to multicast and broadcast messages"
+      property :display_notification, [true, false, nil], description: "Display a notification when firewall blocks certain activity"
+
+      load_current_value do |desired|
+        ps_results = powershell_exec(<<-CODE).result
+          $#{desired.profile} = Get-NetFirewallProfile -Profile #{desired.profile}
+          $#{desired.profile}.Enabled
+          $#{desired.profile}.DefaultInboundAction
+          $#{desired.profile}.DefaultOutboundAction
+          $#{desired.profile}.AllowInboundRules
+          $#{desired.profile}.AllowLocalFirewallRules
+          $#{desired.profile}.AllowLocalIPsecRules
+          $#{desired.profile}.AllowUserApps
+          $#{desired.profile}.AllowUserPorts
+          $#{desired.profile}.AllowUnicastResponseToMulticast
+          $#{desired.profile}.NotifyOnListen
+        CODE
+        profile_enabled case ps_results[0]; when 0 then false; when 1 then true; end
```

Comment:
I think we can do this a bit differently.  Properties default to nil if not set so we can set them to true / false and not set them at all otherwise. In that case current_resource.WHATEVER should be nil. This would allow you to remove nil as an allowed value from all of these properties and allows removing these inline case statements. You might want to just create a simple method to return true/false from 1/2. It would really help the readability here.